backend/infrahub/git/repository.py

```diff
@@ -170,7 +172,11 @@ async def merge(self, source_branch: str, dest_branch: str, push_remote: bool =
         commit = self.get_commit_value(branch_name=source_branch, remote=False)
 
         try:
-            repo.git.merge(commit)
+            if config.SETTINGS.git.allow_explicit_merge_commit:
+                user_name = config.SETTINGS.git.user_name or get_git_user_config(repo)[0]
+                repo.git.merge(commit, "--no-ff", m=f"Merged by Infrahub by {user_name}")
```

Comment:
For now I think we should just go with `repo.git.merge(commit, "--no-ff", m=f"Merged by Infrahub by Infrahub")`

If we'll have a username here it would be the user within Infrahub that merged the branch, or proposed change and not the user set within the Infrahub configuration. The goal of this epic is to add this none ff commit and we haven't gotten a clear answer with regards to what the commit message should be at this point. So let's just keep it simple for now.